### ISSUE TITLE
MA0075: false positive for implicit `ToString()` on char

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#516]
### Version of the Meziantou.Analyzer NuGet package

2.0.45

### Rule Identifier

MA0075

### Target Framework

net48

### C# Language version

C# 11

### Description

**MA0075** is reported for implicit `ToString()` on non-literal (?) char.

### Reproduction Steps

Minimal code:

```c#
char c = '!';
return "Just works" + char.ToLower(c, CultureInfo.InvariantCulture);
```

### Other information

It works fine if you use the **char** directly:

```c#
char c = '!';
return "Just works" + c;
```

**Comment by meziantou** [comment]
I can reproduce the error using the following test:

````c#
    [Fact]
    [Trait("Issue", "https://github.com/meziantou/Meziantou.Analyzer/issues/516")]
    public async Task ConcatNoDiagnostic_Char()
    {
        var sourceCode = """
class Test
{
    void A()
    {
        var c = '!';
        _ = "abc" + char.ToLower(c, System.Globalization.CultureInfo.InvariantCulture);
    }
}
""";
        await CreateProjectBuilder()
              .WithSourceCode(sourceCode)
              .ValidateAsync();
    }
````

I think the fix should be at https://github.com/meziantou/Meziantou.Analyzer/blob/83f0d0e62ec95a8fdebb42b8237eefd3e9478d60/src/Meziantou.Analyzer/Internals/CultureSensitiveFormattingContext.cs#L103. It should check the return type instead of return true (`return IsCultureSensitiveType(invocation.Type, options);`)

**Comment by drieseng** [comment]
@meziantou, I hereby grant you permission to commit the fix :-)

**Comment by drieseng** [comment]
@meziantou, thanks again!
